src/lib/components/Textfield.react.js

```diff
@@ -0,0 +1,186 @@
+  fireEvent?: () => void,
+  /** If true, the input will take up the full width of its container */
+  fullWidth?: boolean,
+  /** The helper text content */
+  helperText?: Node,
+  /** Element ID */
+  id: string,
+  /** If true, a textarea element will be rendered instead of an input */
+  multiline?: boolean,
+  /** If true, the label is displayed as required and the input element` will be required */
+  required?: boolean,
+  /** Dash callback to update props on the server. */
+  setProps?: () => void,
+  /** 'medium' | 'small' */
+  size?: string,
+  /** Override the inline-styles of the root element */
+  style?: Object,
+  /** Type of the input element. It should be a valid HTML5 input type */
+  type?: string,
+  /** The value of the input element, required for a controlled component */
+  value?: any,
+  /** 'filled' | 'outlined' | 'standard' */
+  variant?: string,
+};
+
+type State = {
+  disabled: boolean,
+  value: any,
+};
+
+const defaultProps = {
+  dashCallbackDelay: 500,
+  autoFocus: false,
+  classes: {},
+  className: '',
+  color: 'primary',
+  disabled: false,
+  error: false,
+  fireEvent: () => {},
+  fullWidth: false,
+  helperText: '',
+  multiline: false,
+  required: false,
+  setProps: () => {},
+  size: 'medium',
+  style: {},
+  type: 'text',
+  value: '',
+  variant: 'standard'
+};
+
+/** Material UI TextField component */
+export default class TextField extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      disabled: this.props.disabled,
+      value: this.props.value,
+    };
+    /** _.debounce used to provide delay in callback to avoid firing callback every
+     * time user input changes - waits this.props.dashCallbackDelay ms to fire callback */
+    this.updateTextProps = _.debounce(this._updateTextProps, this.props.dashCallbackDelay);
+  }
+
+  handleChange = (event) => {
+    this.setState({value: event.target.value});
+
+    if (typeof this.props.setProps === 'function') {
+      this.props.setProps({value: event.target.value});
+    }
+  };
+
+  /**
+   * executes setProps function with value to update value in props;
+   * fires Dash callback event
+   * @param value
+   * @private
+   */
+  _updateTextProps = (value) => {
+    const { setProps } = this.props;
+
+    if (typeof setProps === 'function') {
+      setProps({value});
+    }
+
+    if (this.props.fireEvent) {
+      this.props.fireEvent({event: 'change'});
+    }
+  };
+
+
+  render() {
+    const {
+      id,
+      autoFocus,
+      className,
+      color,
+      disabled,
+      error,
+      helperText,
+      multiline,
+      required,
+      size,
+      style,
+      type,
+      value,
+      variant,
+    } = this.props;
+    this.handleChange = this.handleChange.bind(this)
+    this.updateTextProps = this.updateTextProps.bind(this)
+    return (
+      <div className={className}>
+        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
+          <MuiTextField
```

Comment:
I think you'll also need to set the `onChange` prop for this compnent. Something like this probably:
```javascript
          onChange={(event) => {
              this.setState({value: event.target.value})
              this.updateTextProps(event.target.value)}}
```

Reply:
oh, this definitely resolves problems that I found trying to use the component.